src/kcas.ml

```diff
@@ -174,8 +174,8 @@ let rec update_no_alloc backoff loc state set_after =
     let backoff = Backoff.once backoff in
     update_no_alloc backoff loc state set_after
 
-let is_obstruction_free casn =
-  Atomic.get casn == (Mode.obstruction_free :> status)
+let is_obstruction_free casn loc =
+  Atomic.get casn == (Mode.obstruction_free :> status) && loc.lock_mode == Mode.lock_free
```

Comment:
Should the comparison be against `Mode.obstruction_free`?

Reply:
Yes , It should be against Mode.obstruction_free.😅